contracts/OracleService.aes

```diff
@@ -104,16 +102,19 @@ contract OracleService =
     let check_query = (oracle, query) => switch(Oracle.get_answer(oracle, query))
       Some(response) => Some(response == Address.to_str(account))
       None => None
-    [ check_query(oracle, query) | (oracle, query) <- state.oracle_queries[url] ]
+    [ check_query(oracle, query) | (oracle, query) <- state.oracle_queries[url], if (check_oracle(oracle)) ]
 
   function
-    estimate_query_fee_iterate : (int, list((int * oracle_type))) => int
+    estimate_query_fee_iterate : (int, list(oracle_type)) => int
     estimate_query_fee_iterate(fee_acc, [])      = fee_acc
-    estimate_query_fee_iterate(fee_acc, o :: os) = switch(o)
-      (_, oracle) => estimate_query_fee_iterate(fee_acc + Oracle.query_fee(oracle), os)
+    estimate_query_fee_iterate(fee_acc, o :: os) = estimate_query_fee_iterate(fee_acc + Oracle.query_fee(o), os)
```

Comment:
What is the behavior here, having 3 functions with the same name?

Reply:
its pattern matching, see https://github.com/aeternity/aesophia/releases/tag/v4.2.0 changelog